src/tables/HVAR.js

```diff
@@ -0,0 +1,78 @@
+import r from 'restructure';
+import {resolveLength} from 'restructure/src/utils';
+
+
+let F2DOT14 = new r.Fixed(16, 'BE', 14);
+let RegionAxisCoordinates = new r.Struct({
+  startCoord: F2DOT14,
+  peakCoord: F2DOT14,
+  endCoord: F2DOT14
+});
+
+let VariationRegionList = new r.Struct({
+  axisCount: r.uint16,
+  regionCount: r.uint16,
+  variationRegions: new r.Array(new r.Array(RegionAxisCoordinates, 'axisCount'), 'regionCount')
+});
+
+let DeltaSet = new r.Struct({
+  shortDeltas: new r.Array(r.int16, t => t.parent.shortDeltaCount),
+  regionDeltas: new r.Array(r.int8, t => t.parent.regionIndexCount - t.parent.shortDeltaCount),
+  deltas: t => t.shortDeltas.concat(t.regionDeltas)
+});
+
+let ItemVariationData = new r.Struct({
+  itemCount: r.uint16,
+  shortDeltaCount: r.uint16,
+  regionIndexCount: r.uint16,
+  regionIndexes: new r.Array(r.uint16, 'regionIndexCount'),
+  deltaSets: new r.Array(DeltaSet, 'itemCount')
+});
+
+let ItemVariationStore = new r.Struct({
+  format: r.uint16,
+  variationRegionList: new r.Pointer(r.uint32, VariationRegionList),
+  variationDataCount: r.uint16,
+  itemVariationData: new r.Array(new r.Pointer(r.uint32, ItemVariationData), 'variationDataCount')
+});
+
+// TODO: add this to restructure
+class VariableSizeNumber {
+  constructor(size) {
+    this._size = size;
+  }
+
+  decode(stream, parent) {
+    switch (this.size(0, parent)) {
```

Comment:
As potentially faster alternative, you could rewrite this switch as a lookup - for a frequently used operation, it might make just enough of a difference to be appreciable (although some perf testing would have to be done to confirm that):

```
const READERS = [
  'readUInt8',
  'readUInt16BE',
  'readUInt24BE',
  'readUInt32BE'
];

decode(stream, parent) {
  return stream[READERS[this.size(0,parent))]]();
}
```